src/codegen/sys/tests.rs

```diff
@@ -233,27 +246,37 @@ fn generate_constant_c(env: &Env, path: &Path, w: &mut dyn Write) -> io::Result<
         w,
         "{}",
         r####"
-int main() {
-    printf(_Generic((ABI_CONSTANT_NAME),
-                    char *: "###gir test###%s###gir test###\n",
-                    const char *: "###gir test###%s###gir test###\n",
-                    char: "###gir test###%c###gir test###\n",
-                    signed char: "###gir test###%hhd###gir test###\n",
-                    unsigned char: "###gir test###%hhu###gir test###\n",
-                    short int: "###gir test###%hd###gir test###\n",
-                    unsigned short int: "###gir test###%hu###gir test###\n",
-                    int: "###gir test###%d###gir test###\n",
-                    unsigned int: "###gir test###%u###gir test###\n",
-                    long: "###gir test###%ld###gir test###\n",
-                    unsigned long: "###gir test###%lu###gir test###\n",
-                    long long: "###gir test###%lld###gir test###\n",
-                    unsigned long long: "###gir test###%llu###gir test###\n",
-                    double: "###gir test###%f###gir test###\n",
-                    long double: "###gir test###%ld###gir test###\n"),
-           ABI_CONSTANT_NAME);
-    return 0;
-}"####
-    )
+#define PRINT_CONSTANT(CONSTANT_NAME) \
+    printf("%s;", #CONSTANT_NAME); \
+    printf(_Generic((CONSTANT_NAME), \
```

Comment:
OOC, does MSVC support that? It's "new" in C11.

Also we should probably make sure to pass `--std=c11` or so to the compiler.

Reply:
Do you mean `_Generic`? note that it was there before as well. I have not started to worry about MSVC yet :)

Reply:
Yes